xla/tools/multihost_hlo_runner/hlo_runner_main.cc

```diff
@@ -382,10 +407,24 @@ int main(int argc, char** argv) {
       tsl::Flag("profile_execution", &opts.profile_execution,
                 "If set, we will profile the execution and print the results."),
       tsl::Flag("xla_gpu_dump_xspace_to", &opts.xla_gpu_dump_xspace_to,
-                "A directory to dump xspace data for GPU profiling.")};
+                "A directory to dump xspace data for GPU profiling."),
+      // This option is not used during parsing, but it is added here for
+      // documentation, and for ensuring that the parser knows this should be
+      // ignored if present.
+      tsl::Flag("debug_options_file", &opts.debug_options_filename,
+                "A file containing debug options to be passed to the HLO "
+                "module. The file should contain a serialized DebugOptions "
+                "proto message. The order of precedence: command line flags > "
+                "XLA_FLAGS > debug_options_file > default flags.")};
 
   xla::AppendDebugOptionsFlags(&flag_list);
 
+  auto debugOptionsFilename = GetDebugOptionsFileName(argc, argv);
```

Comment:
It's hacky to parse argc and argv like this outside absl. An alternative might be to call `tsl::Flags::Parse` twice: once to look for the `--debug_options_file` flag, and again to parse everything else afterwards. But this admittedly is hacky as well, and I cannot think of a better way. 

You can keep this for now but I'll keep thinking if there's a better way.

Reply:
Marking this as "resolved" for now.